flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksUtils.java

```diff
@@ -297,10 +297,29 @@ public StarRocksColumn.Builder visit(DoubleType doubleType) {
 
         @Override
         public StarRocksColumn.Builder visit(DecimalType decimalType) {
-            builder.setDataType(DECIMAL);
+            // StarRocks is not support Decimal as primary key, so decimal should be cast to INT,
+            // BIGINT, LARGEINT or VARHCAR.
+            if (!isPrimaryKeys) {
+                builder.setDataType(DECIMAL);
+                builder.setColumnSize(decimalType.getPrecision());
+                builder.setDecimalDigits(decimalType.getScale());
+            } else if (decimalType.getPrecision() < 10 && decimalType.getScale() == 0) {
```

Comment:
`getScale() == 0` check was repeated 3 times. Maybe check it first, and fallback to `VARCHAR` if such DecimalType were used?